src/observables/OptionStream.ts

```diff
@@ -0,0 +1,42 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Options } from "../omnisharp/options";
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { vscode } from "../vscodeAdapter";
+import 'rxjs/add/operator/take';
+import Disposable, { IDisposable } from "../Disposable";
+
+export default class OptionStream {
```

Comment:
The API that this class isn't really stream-y. Do we really need to have "stream" in the name?

Reply:
Should we break this into 2 concepts? One thing is the "optionstream" that subscribes to the VS Code option system and and acts like an Observable. There are then 2 observers: one provides the Options() functionality currently described in this file, and the options changed observer you're going to add later?